app/src/main/kotlin/com/adriangl/pokeapi_mvvm/pokemonlist/PokemonListViewModel.kt

```diff
@@ -48,10 +49,10 @@ class PokemonListViewModel(app: Application) : RxAndroidViewModel(app), KodeinAw
             .subscribe {
                 if (it.isSuccess) {
                     val movesList = pokeStore.state.pokemonList!!
-                        .flatMap {
-                            it.moves.subList(0, 4.coerceAtMost(it.moves.size))
-                        } // Only 4 movements
-                        .distinctBy { it.move.name }.map { it.move.name }
+                        .flatMap { pokemon ->
+                            pokemon.moves.subList(0, 4.coerceAtMost(pokemon.moves.size))
+                        } // At most 4 movements
+                        .map { it.move.name }.distinct()
```

Comment:
I tweaked this line a little bit since the `distinctBy` and then `map` to get only names could be reduced to `map` to names and then `distinct`

Reply:
Looks good to me